PacketDefinitions420/PacketNotifier.cs

```diff
@@ -3811,7 +3816,12 @@ void NotifyEnterTeamVision(IGameObject obj, TeamId team, int userId = 0, GamePac
                 }
                 var healthbarPacket = ConstructEnterLocalVisibilityClientPacket(obj);
                 //TODO: try to include it to packets too?
-                var us = new UpdateStats(u.Replication, false);
+                var us = new OnReplication(){
```

Comment:
Instead of calling `new OnReplication` here, why not use the replication function itself?

Reply:
I don't quite understand. I can add another function like `ConstructOnReplication(IGameObject for, bool partial)`. But you will still have to get rid of it later, when the transition is made from creating a packet for each unit to creating one packet for all units.

Reply:
How do you not understand this? I have known this for ages, you piece of shit.